src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -1179,6 +1181,394 @@ default boolean isFromGuild()
+     *
+     * @param  id
+     *         The string id of the reaction emoji.
+     *
+     * @throws net.dv8tion.jda.api.exceptions.InsufficientPermissionException
+     *         If the MessageChannel this message was sent in was a {@link net.dv8tion.jda.api.entities.TextChannel TextChannel} and the
+     *         logged in account does not have {@link net.dv8tion.jda.api.Permission#MESSAGE_HISTORY Permission.MESSAGE_HISTORY} in the channel.
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided id is not a valid snowflake.
+     *
+     * @return The {@link net.dv8tion.jda.api.entities.MessageReaction.ReactionEmote ReactionEmote} of this message or null if not present.
+     *
+     * @since 4.1.0
+     */
+    @Nullable
+    @CheckReturnValue
+    MessageReaction.ReactionEmote getReactionById(@Nonnull String id);
+
+    /**
+     * This obtains the {@link net.dv8tion.jda.api.entities.MessageReaction.ReactionEmote ReactionEmote} for the given reaction id on this message.
+     *
+     * <p>Messages store reactions by keeping a list of reaction names.
+     *
+     * <p>An instance of the related {@link net.dv8tion.jda.api.entities.MessageReaction.ReactionEmote ReactionEmote} can be
+     * obtained through this method by using the emoji's id.
+     *
+     *
+     * <p>The following {@link net.dv8tion.jda.api.requests.ErrorResponse ErrorResponses} are possible:
+     * <ul>
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#MISSING_ACCESS MISSING_ACCESS}
+     *     <br>The request was attempted after the account lost access to the {@link net.dv8tion.jda.api.entities.TextChannel TextChannel}
+     *         due to {@link net.dv8tion.jda.api.Permission#MESSAGE_READ Permission.MESSAGE_READ} being revoked
+     *     <br>Also can happen if the account lost the {@link net.dv8tion.jda.api.Permission#MESSAGE_HISTORY Permission.MESSAGE_HISTORY}</li>
+     *
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#UNKNOWN_MESSAGE UNKNOWN_MESSAGE}
+     *         The reaction request was attempted after the Message had been deleted.</li>
+     * </ul>
+     *
+     * @param  id
+     *         The long id of the reaction emoji.
+     *
+     * @throws net.dv8tion.jda.api.exceptions.InsufficientPermissionException
+     *         If the MessageChannel this message was sent in was a {@link net.dv8tion.jda.api.entities.TextChannel TextChannel} and the
+     *         logged in account does not have {@link net.dv8tion.jda.api.Permission#MESSAGE_HISTORY Permission.MESSAGE_HISTORY} in the channel.
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided id is not a valid snowflake.
+     *
+     * @return The {@link net.dv8tion.jda.api.entities.MessageReaction.ReactionEmote ReactionEmote} of this message or null if not present.
+     *
+     * @since 4.1.0
+     */
+    @Nullable
+    @CheckReturnValue
+    MessageReaction.ReactionEmote getReactionById(@Nonnull Long id);
```

Comment:
Use primitive longs

Reply:
Was using those before but I changed them due to the compiler giving off an annotation warning over primitives, will switch them back and remove the annotations.